src/sentry/static/sentry/app/components/accordion.jsx

```diff
@@ -0,0 +1,81 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import styled from 'react-emotion';
+import InlineSvg from 'app/components/inlineSvg';
+import space from 'app/styles/space';
+import {Flex} from 'grid-emotion';
+import Badge from 'app/components/badge';
+import Button from 'app/components/buttons/button';
+
+class AccordionButton extends React.Component {
+  static propTypes = {
+    label: PropTypes.string.isRequired,
+    cutoff: PropTypes.number,
+    children: PropTypes.arrayOf(PropTypes.node).isRequired,
+  };
+
+  static defaultProps = {
+    cutoff: 0,
+  };
+
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      isOpen: false,
+    };
+  }
+
+  handleClick(e) {
+    e.preventDefault();
+    this.setState({isOpen: !this.state.isOpen});
+  }
+
+  render() {
+    const {label, cutoff, children} = this.props;
+
+    if (children.length <= cutoff + 1) return children;
+
+    return (
+      <div>
+        {children.slice(0, cutoff)}
+        <StyledButton onClick={e => this.handleClick(e)} open={this.state.isOpen}>
```

Comment:
Let's bind `this.handleClick` as a class property, e.g.
```
  handleClick = (e) => {
    e.preventDefault();
    this.setState({isOpen: !this.state.isOpen});
  }
```

Reply:
ok THANK YOU I actually meant to ask about this.

It has always baffled me a little bit as to when and why `this` is undefined in react classes. Your suggestion makes a lot of sense (it would be similar to passing in an anonymous function like I did) but i've never understood why defining it as a class method and just passing it in like `onClick={handleClick}` gets me `undefined`.

Reply:
Yeah it's because `this` context isn't bound (there are like 10 other ways to do it, but we generally use class properties).